packages/astro/src/core/app/index.ts

```diff
@@ -131,8 +134,31 @@ export class App {
 		const url = new URL(request.url);
 		// ignore requests matching public assets
 		if (this.#manifest.assets.has(url.pathname)) return undefined;
-		const pathname = prependForwardSlash(this.removeBase(url.pathname));
-		const routeData = matchRoute(pathname, this.#manifestData);
+		let pathname;
+		if (this.#manifest.i18n && this.#manifest.i18n.routingStrategy === 'domain') {
+			const host = request.headers.get('X-Forwarded-Host');
+			if (host) {
+				try {
+					const locale = this.#manifest.i18n.domainLookupTable[host];
```

Comment:
Does the `domainLookupTable` contain the protocol? Does the `host` contain it? If not then they may not match here.

Reply:
Good call out. I believe it doesn't have the protocol